src/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -27,7 +28,7 @@ public abstract unsafe class AbstractPager : IDisposable, ILowMemoryHandler
 
         public static ConcurrentDictionary<string, uint> PhysicalDrivePerMountCache = new ConcurrentDictionary<string, uint>();
 
-        protected static readonly object WorkingSetIncreaseLocker = new object();
+        protected static readonly ReaderWriterLockSlim WorkingSetIncreaseLocker = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
```

Comment:
How much contention are we likely to actually get? Not worth the trouble , I thik

Reply:
Startup of many encrypted databases can have a lot of contention... but for most of the normal operations, there shouldn't be much.

Reply:
Recently we saw the issue in our own system for that scenario - after the restart of the production server, once databases were loaded

Reply:
The problem is that you are not actually using the reader / writer mode. 
You have a single entry at any given point anyway. Better to use a simpler lock